### DISCUSSION TITLE
JWT modify

### DISCUSSION THREAD

**Discussion by marcoarnulfo** [pocketbase/pocketbase#4721] [Q&A]
Simple question: 
Is there a "simple" way to add some fields to the token? Like a "isAdmin"?

Also, I must say, great job with PocketBase!

**Comment by ganigeorgiev** [comment]
No, the default auth token payload is not customizable as we usually keep only record fields that are not expected to change from one request to another (ex. `id`).

The question lack details to understand why you want to do that, but note that even if you create your own custom auth token (_see the [tokens helpers](https://github.com/pocketbase/pocketbase/blob/8264ead4debff819275b3625679a6627abafb636/tokens/record.go#L13) for an example_) you still will have to verify the token signature server-side (_PocketBase tokens have unique secret for each user since it is constructed from 2 parts - common stored in the app settings and user unique part stored as `tokenKey` for the specific user_).

**Reply by marcoarnulfo** [reply]
Including a value like isAdmin in the token (or something similar) would allow me to have users with special permissions

**Reply by ganigeorgiev** [reply]
Sorry but it is still not clear what you are trying to do - how this field will be set, who will have control to change it, how and when old tokens will be invalidated in case `isAdmin` changes, etc.

Again, even if you add the field to the token you can't trust the value unless you first verify the token signature and in order to do that you'll need to fetch the user anyway to construct the token secret, which means that you may as well simply store your flag value as part of the user record that will be returned on `getOne` or `authRefresh` calls.

In any case for permissions and access controls, you could check [API rules and filters docs](https://pocketbase.io/docs/api-rules-and-filters/).

**Reply by marcoarnulfo** [reply]
Ye, I did it like this:

`const userDb = await pb.collection('users').getOne(userId)`
`const userRole = userDb.admin`
`return userRole === PermissionsType.ADMIN`

Thank you so much for your time!
